bin/parse_pom_xml

```diff
@@ -22,17 +22,28 @@ if parsed_file.errors.size.positive?
   exit(1)
 end
 
-dependencies = []
+properties = {}
+# Parse contents within each <properties> tag
+nodes = parsed_file.css('properties')
+nodes.each do |node|
+  childrens = node.children
+  childrens.each do |children|
+    properties["${#{children.name}}"] = children.text
+  end
+end
 
+dependencies = []
 # Parse contents within each <dependency> tag
 parsed_file.css('dependency').each do |dependency|
   group_id = dependency.at('groupId')
   artifact_id = dependency.at('artifactId')
   version = dependency.at('version')
+  version = properties.dig(version.children.to_s) || version.children.to_s unless version.nil?
```

Comment:
could you provide a pom example of why this is needed?

Reply:
In pom.xml, dependencies are sometimes defined as - 
```
<dependency>
    <groupId>ch.qos.logback</groupId>
    <artifactId>logback-classic</artifactId>
    <version>${logback.version}</version>
</dependency>
```
In this case we used to report version as `${logback.version}` which is incorrect and can cause issues. The code highlighted is a fix for it.
These variables are defined in `<properties>` tag
```
<properties>
<logback.version>1.2.10</logback.version>
</properties>
```
So I parse it and try to map it when building the dependencies array to get more accurate representation.

Reply:
could you add that sample xml into the code as a comment?